lab_2_tokenize_by_bpe/main.py

```diff
@@ -38,22 +42,21 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
-    if not isinstance(text, str) or not isinstance(end_of_word, str) or not (
-            isinstance(start_of_word, str) or start_of_word is None):
+    if not isinstance(text, str) or not isinstance(end_of_word, str)\
+            or not (isinstance(start_of_word, str) or start_of_word is None):
         return None
 
-    dict_frequencies = {}
-
-    splitted_text = text.split()
-    for i in set(splitted_text):
-        word = prepare_word(i, start_of_word, end_of_word)
-        if not word:
+    word_list = text.split()
+    freq_dict = {}
+    for word in word_list:
+        freq = word_list.count(word)
```

Comment:
count count function for the same values many times - try to accelerate

Reply:
improved